File: backend/app/Http/Controllers/LabelController.php
<?php

namespace App\Http\Controllers;

use App\Models\Label;
use App\Http\Requests\StoreLabelRequest;
use App\Http\Requests\UpdateLabelRequest;
use Illuminate\Http\Request;
use Illuminate\Http\Response;

class LabelController extends Controller
{
    //  GET ALL
    public function index(Request $request)
    {
        $userId = $request->user()?->id;

        return Label::query()
            ->when($userId !== null, fn ($query) => $query->where('user_id', $userId))
            ->get();
    }

    // CREATE
    public function store(StoreLabelRequest $request)
    {
        $userId = $request->user()?->id;
        if ($userId === null) {
            return response()->json(['message' => 'Unauthenticated.'], 401);
        }

        $label = Label::create([
            'user_id' => $userId,
            'name' => $request->validated('name'),
        ]);

        return response()->json($label, 201);
    }

    //  UPDATE
    public function update(UpdateLabelRequest $request, $id)
    {
        $label = Label::findOrFail($id);
        $userId = $request->user()?->id;

        if ($userId === null) {
            return response()->json(['message' => 'Unauthenticated.'], 401);
        }

        if ((int) $label->user_id !== (int) $userId) {
            return response()->json(['message' => 'Forbidden.'], 403);
        }

        $label->update([
            'name' => $request->validated('name') ?? $label->name,
        ]);

        return $label;
    }

    // DELETE
    public function destroy(Request $request, $id)
    {
        $label = Label::findOrFail($id);
        $userId = $request->user()?->id;

        if ($userId === null) {
            return response()->json(['message' => 'Unauthenticated.'], 401);
        }

        if ((int) $label->user_id !== (int) $userId) {
            return response()->json(['message' => 'Forbidden.'], 403);
        }
        $label->delete();

        return response()->json([
            'message' => 'Deleted'
        ]);
    }
}

File: backend-spring/src/main/java/com/jotdown/api/controller/LabelController.java
package com.jotdown.api.controller;

import com.jotdown.api.dto.request.StoreLabelRequest;
import com.jotdown.api.dto.request.UpdateLabelRequest;
import com.jotdown.api.dto.response.LabelResponse;
import com.jotdown.api.entity.User;
import com.jotdown.api.security.CurrentUser;
import com.jotdown.api.service.LabelService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/labels")
@RequiredArgsConstructor
public class LabelController {

    private final LabelService labelService;

    @GetMapping
    public ResponseEntity<List<LabelResponse>> index(@CurrentUser User currentUser) {
        List<LabelResponse> labels = labelService.listLabels(currentUser);
        return ResponseEntity.ok(labels);
    }

    @PostMapping
    public ResponseEntity<LabelResponse> store(
            @CurrentUser User currentUser,
            @Valid @RequestBody StoreLabelRequest request) {
        LabelResponse label = labelService.createLabel(request, currentUser);
        return ResponseEntity.status(HttpStatus.CREATED).body(label);
    }

    @PutMapping("/{id}")
    public ResponseEntity<LabelResponse> update(
            @CurrentUser User currentUser,
            @PathVariable("id") Long id,
            @Valid @RequestBody UpdateLabelRequest request) {
        LabelResponse label = labelService.updateLabel(id, request, currentUser);
        return ResponseEntity.ok(label);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, String>> destroy(
            @CurrentUser User currentUser,
            @PathVariable("id") Long id) {
        labelService.deleteLabel(id, currentUser);
        Map<String, String> response = new HashMap<>();
        response.put("message", "Deleted");
        return ResponseEntity.ok(response);
    }
}
